### PR TITLE
Add backup backup collection op

### PR DESCRIPTION
**Pull request by Enjection**
### Changelog entry 

...

### Changelog category 

* Not for changelog (changelog entry is not required)

### Additional information

...

### CODE REVIEW

File: ydb/core/tx/schemeshard/schemeshard__operation_backup_backup_collection.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+#define LOG_I(stream) LOG_INFO_S(context.Ctx, NKikimrServices::FLAT_TX_SCHEMESHARD, "[" << context.SS->TabletID() << "] " << stream)
+#define LOG_N(stream) LOG_NOTICE_S(context.Ctx, NKikimrServices::FLAT_TX_SCHEMESHARD, "[" << context.SS->TabletID() << "] " << stream)
+#define RETURN_RESULT_UNLESS(x) if (!(x)) return result;
+
+namespace NKikimr::NSchemeShard {
+
+TString ToX509String(const TInstant& datetime) {
+    return datetime.FormatLocalTime("%Y%m%d%H%M%SZ");
+}
```
Comment (ijon) [inline ydb/core/tx/schemeshard/schemeshard__operation_backup_backup_collection.cpp:15]: Doesn't seem to be used in the module.

File: ydb/core/tx/schemeshard/schemeshard__operation_backup_backup_collection.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
+#include "schemeshard__operation_common.h"
+#include "schemeshard__operation_create_cdc_stream.h"
+#include "schemeshard_impl.h"
+
+#include <ydb/core/tx/schemeshard/backup/constants.h>
+
+#define LOG_I(stream) LOG_INFO_S(context.Ctx, NKikimrServices::FLAT_TX_SCHEMESHARD, "[" << context.SS->TabletID() << "] " << stream)
+#define LOG_N(stream) LOG_NOTICE_S(context.Ctx, NKikimrServices::FLAT_TX_SCHEMESHARD, "[" << context.SS->TabletID() << "] " << stream)
+#define RETURN_RESULT_UNLESS(x) if (!(x)) return result;
```
Comment (ijon) [inline ydb/core/tx/schemeshard/schemeshard__operation_backup_backup_collection.cpp:9]: Not used in the module.
(And I would consider its usage unjustified anyway.)

File: ydb/core/tx/schemeshard/schemeshard__op_traits.h [hunk 3]
```diff
@@ -290,4 +299,65 @@ struct TSchemeTxTraits<NKikimrSchemeOp::EOperationType::ESchemeOpCreateBackupCol
...
+
+        for (const auto& item : bc->Description.GetExplicitEntryList().GetEntries()) {
+            Y_ABORT_UNLESS(prefixLen <= item.GetPath().length());
+            TString itemName = item.GetPath().substr(prefixLen, item.GetPath().size() - prefixLen);
+            auto pos = itemName.rfind("/");
+            if (pos != std::string::npos) {
+                itemName.resize(pos);
+                collectionPaths.emplace(JoinPath({targetDir, itemName}));
+            }
+        }
```
Comment (ijon) [inline ydb/core/tx/schemeshard/schemeshard__op_traits.h:344]: Paths work here seems to be brittle. Is it possible to express it via splits and joins?

File: ydb/core/tx/schemeshard/schemeshard__op_traits.h [hunk 4]
```diff
@@ -290,4 +299,65 @@ struct TSchemeTxTraits<NKikimrSchemeOp::EOperationType::ESchemeOpCreateBackupCol
     constexpr inline static bool CreateDirsFromName = true;
 };
 
+template <>
+struct TSchemeTxTraits<NKikimrSchemeOp::EOperationType::ESchemeOpBackupBackupCollection> : public TSchemeTxTraitsFallback {
+    static std::optional<THashMap<TString, THashSet<TString>>> GetRequiredPaths(const TTxTransaction& tx, const TOperationContext& context) {
```
Comment (ijon) [inline ydb/core/tx/schemeshard/schemeshard__op_traits.h:304]: Very specific and non primitive manipulations here.
I would suggest moving `GetRequiredPaths()` definition closer to the operation itself, to `schemeshard__operation_backup_backup_collection.cpp`.

File: ydb/core/tx/schemeshard/ut_backup_collection/ut_backup_collection.cpp [hunk 5]
```diff
@@ -288,4 +288,58 @@ Y_UNIT_TEST_SUITE(TBackupCollectionTests) {
...
+        TTestBasicRuntime runtime;
+        TTestEnv env(runtime, TTestEnvOptions().EnableBackupService(true));
+        ui64 txId = 100;
+
+        SetupLogging(runtime);
+
+        PrepareDirs(runtime, env, txId);
+
+        TestBackupBackupCollection(runtime, ++txId, "/MyRoot",
+            R"(Name: ".backups/collections/)" DEFAULT_NAME_1 R"(")",
```
[the remark this answers is not in the document]
Comment (Enjection) [reply ydb/core/tx/schemeshard/ut_backup_collection/ut_backup_collection.cpp:302]: We discussed it earlier on a meeting, so it seems that we should just acknowledge it

[the remark this answers is not in the document]
Comment (Enjection) [reply ydb/core/tx/schemeshard/ut_backup_collection/ut_backup_collection.cpp:302]: By the way, for now we don't even need it to be system, there could be any garbage. The only consequence will be unavailability of backups functionality in certain cases (when `.backups/collections` or just `.backups` is not directory)

**Review by ijon — APPROVED** [review]

**Pull request merged**